### ISSUE TITLE
Training Segnet for defect detections in archived video 

### ISSUE THREAD

**Issue by HamzaYousVision** [alexgkendall/caffe-segnet#90]
Hi, @alexgkendall ,

I inspired from your architecture to design a simpler Encode-decoder network in the aim to detect blotches in archived video. Initially, I want to begin by frame-by-frame without considering the temporal information. The output of my network is two classes for each pixel (blotch or not blotch). But the problem is that after only 20 iterations, I obtain this result 

 Train net output #0: accuracy = 0.996994
I0105 14:57:17.983150  7145 solver.cpp:229]     Train net output #1: loss = 0.00522148 (* 1 = 0.00522148 loss)
I0105 14:57:17.983162  7145 solver.cpp:229]     Train net output #2: per_class_accuracy = 0
I0105 14:57:17.983175  7145 solver.cpp:229]     Train net output #3: per_class_accuracy = 1

I did not understand why the accuracy fr the second class in always 1?
could you help me to figure out my problem?

Many thanks in advance,

Hamza

**Comment by ANIS87** [comment]
problem of class_weighting you need to cheek if you have unbalanced class (for example in image 80% class 1 and 20% class 2)
